File: swift/Samples/basic_card_sample.swift
import FluentCards

func createWelcomeCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Welcome to FluentCards!")
                .withSize(.large)
                .withWeight(.bolder)
                .withHorizontalAlignment(.center)
        }
        .addTextBlock { tb in
            tb.withText("This library helps you create Adaptive Cards using a fluent API.")
                .withWrap(true)
        }
        .addAction { a in
            a.openURL("https://adaptivecards.io").withTitle("Learn More")
        }
        .build()
}

func createNotificationCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addTextBlock { tb in
            tb.withText("Notification")
                .withSize(.medium)
                .withWeight(.bolder)
                .withColor(.attention)
        }
        .addTextBlock { tb in
            tb.withText("You have a new message waiting for you.")
                .withWrap(true)
        }
        .addAction { a in
            a.openURL("https://example.com/messages").withTitle("View Messages")
        }
        .build()
}

func createImageCard() -> Card {
    return AdaptiveCardBuilder()
        .withVersion("1.5")
        .addImage { img in
            img.withURL("https://adaptivecards.io/content/adaptive-card-50.png")
                .withSize(.medium)
                .withHorizontalAlignment(.center)
        }
        .addTextBlock { tb in
            tb.withText("Adaptive Cards")
                .withSize(.large)
                .withWeight(.bolder)
                .withHorizontalAlignment(.center)
        }
        .addTextBlock { tb in
            tb.withText("Platform-agnostic snippets of UI")
                .withWrap(true)
                .withHorizontalAlignment(.center)
        }
        .build()
}

File: rust/samples/src/basic_card_sample.rs
use fluent_cards::*;

pub fn create_welcome_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Welcome to FluentCards!")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder)
                .with_horizontal_alignment(HorizontalAlignment::Center);
        })
        .add_text_block(|tb| {
            tb.with_text("This library helps you create Adaptive Cards using a fluent API.")
                .with_wrap(true);
        })
        .add_action(|a| {
            a.open_url("https://adaptivecards.io")
                .with_title("Learn More");
        })
        .build()
}

pub fn create_notification_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_text_block(|tb| {
            tb.with_text("Notification")
                .with_size(TextSize::Medium)
                .with_weight(TextWeight::Bolder)
                .with_color(TextColor::Attention);
        })
        .add_text_block(|tb| {
            tb.with_text("You have a new message waiting for you.")
                .with_wrap(true);
        })
        .add_action(|a| {
            a.open_url("https://example.com/messages")
                .with_title("View Messages");
        })
        .build()
}

pub fn create_image_card() -> Card {
    AdaptiveCardBuilder::new()
        .with_version("1.5")
        .add_image(|img| {
            img.with_url("https://adaptivecards.io/content/adaptive-card-50.png")
                .with_size(ImageSize::Medium)
                .with_horizontal_alignment(HorizontalAlignment::Center);
        })
        .add_text_block(|tb| {
            tb.with_text("Adaptive Cards")
                .with_size(TextSize::Large)
                .with_weight(TextWeight::Bolder)
                .with_horizontal_alignment(HorizontalAlignment::Center);
        })
        .add_text_block(|tb| {
            tb.with_text("Platform-agnostic snippets of UI")
                .with_wrap(true)
                .with_horizontal_alignment(HorizontalAlignment::Center);
        })
        .build()
}
